cli/cmd/testdata/install_custom_registry.golden

```diff
@@ -2629,7 +2626,7 @@ spec:
         # see https://github.com/grafana/grafana/issues/20096
         - name: GODEBUG
           value: netdns=go
-        image: my.custom.registry/linkerd-io/grafana:install-control-plane-version
+        image: gcr.io/linkerd-io/grafana:install-control-plane-version
```

Comment:
Any idea why this is changed? The `install_test.go` is still using the `my.custom.registry` override as seen [here](https://github.com/linkerd/linkerd2/blob/46d22f8b04b10fd99dd16a9aec840a3d3c6bea9c/cli/cmd/install_test.go#L218-L227).

Reply:
Yeah, This was because we removed using cli flags, to propogate stuff to add-on fields. This means that users who would want to change to a custom registry ( in this case, or use a different image name) would need to update the field using `--addon-config` through `grafana.image.name`

So, In this case, users would require to update that individually. Do you think we should make sure we propogate configuration changes to add-ons if the flag is a global one (like the custom registry one here)?

Reply:
What is the current behaviour for other global variables? If they are inherited by the add-ons, I think we should do the same here.

Reply:
@ihcsim We definitely follow the `global` fields for add-ons too, but My question was around the CLI flags? Currently, Its a bit confusing whether to support them or not. as users can always update fields using add-on level configuration. Should we propogate the configuration through the CLI flags too?

Reply:
Are any other "global" flags repeated in the add-on config? I'd assume add-on config only affects the add-ons, not the entire control plane. If we want to deprecate a CLI flag, we will have to mark it as so, not just remove them.